sdk/javascript/src/utils/arrayHelpers.js

```diff
@@ -0,0 +1,82 @@
+const { BufferView } = require('./BufferView');
+
+const readArrayImpl = (bufferInput, FactoryClass, accessor, shouldContinue) => {
+	const view = new BufferView(bufferInput);
+	const elements = [];
+	let previousElement = null;
+	let i = 0;
+	while (shouldContinue(i, view)) {
+		const element = FactoryClass.deserialize(view.buffer);
+
+		if (0 >= element.size)
+			throw RangeError('element size has invalid size');
+
+		if (accessor && previousElement && accessor(previousElement) >= accessor(element))
+			throw RangeError('elements in array are not sorted');
+
+		elements.push(element);
+		view.shift(element.size);
+
+		previousElement = element;
+		++i;
+	}
+
+	return elements;
+};
+
+const writeArrayImpl = (output, elements, count, accessor = null) => {
+	for (let i = 0; i < count; ++i) {
+		const element = elements[i];
+		if (accessor && 0 < i && accessor(elements[i - 1]) >= accessor(element))
+			throw RangeError('array passed to write_array is not sorted');
+
+		output.write(element.serialize());
+	}
+};
+
+const arrayHelpers = {
+	alignUp: (size, alignment) => Math.floor((size + alignment - 1) / alignment) * alignment,
+
+	// note: this method is used only for '__FILL__' type arrays
+	// this loop assumes properly sliced buffer is passed and that there's no additional data.
+	readArray: (bufferInput, FactoryClass, accessor = null) =>
+		readArrayImpl(bufferInput, FactoryClass, accessor, (_, view) => 0 < view.buffer.length),
+
+	readArrayCount: (bufferInput, FactoryClass, count, accessor = null) =>
+		readArrayImpl(bufferInput, FactoryClass, accessor, (index, _) => count > index),
+
+	readVariableSizeElements: (bufferInput, FactoryClass, alignment) => {
+		const view = new BufferView(bufferInput);
+		const elements = [];
+		while (0 < view.buffer.length) {
+			const element = FactoryClass.deserialize(view.buffer);
+
+			if (0 >= element.size)
+				throw RangeError('element size has invalid size');
+
+			elements.push(element);
+
+			const alignedSize = arrayHelpers.alignUp(element.size, alignment);
+			if (alignedSize > view.buffer.length)
+				throw RangeError('unexpected buffer length');
+
+			view.shift(alignedSize);
+		}
+
+		return elements;
+	},
+
+	writeArray: (output, elements, accessor = null) => writeArrayImpl(output, elements, elements.length, accessor),
```

Comment:
unless you have good reason, you probably want `undefined` instead of `null` here